src/main/java/com/fabriccommunity/spookytime/block/SpookyCactusBlock.java

```diff
@@ -0,0 +1,141 @@
+package com.fabriccommunity.spookytime.block;
+
+import java.util.Iterator;
+import java.util.Random;
+
+import com.fabriccommunity.spookytime.entity.SpookyCactusEntity;
+import com.fabriccommunity.spookytime.registry.SpookyBlocks;
+import com.fabriccommunity.spookytime.registry.SpookyEntities;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockPlacementEnvironment;
+import net.minecraft.block.BlockRenderLayer;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.Material;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityContext;
+import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.state.StateFactory;
+import net.minecraft.state.property.IntProperty;
+import net.minecraft.state.property.Properties;
+import net.minecraft.tag.FluidTags;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.shape.VoxelShape;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.ViewableWorld;
+import net.minecraft.world.World;
+
+public class SpookyCactusBlock extends Block {
+	public static final IntProperty AGE;
+	protected static final VoxelShape COLLISION_SHAPE;
+	protected static final VoxelShape OUTLINE_SHAPE;
+
+	static {
+		AGE = Properties.AGE_15;
+		COLLISION_SHAPE = Block.createCuboidShape(1.0D, 0.0D, 1.0D, 15.0D, 15.0D, 15.0D);
+		OUTLINE_SHAPE = Block.createCuboidShape(1.0D, 0.0D, 1.0D, 15.0D, 16.0D, 15.0D);
+	}
+	
+	public SpookyCactusBlock(Block.Settings settings) {
+		super(settings);
+		this.setDefaultState(this.stateFactory.getDefaultState().with(AGE, 0));
+	}
+
+	public void onScheduledTick(BlockState state, World world, BlockPos pos, Random random) {
+		if (!state.canPlaceAt(world, pos)) {
+			world.breakBlock(pos, true);
+		} else {
+			BlockPos pos2 = pos.up();
+			if (world.isAir(pos2)) {
+				int height = 1;
+				while(world.getBlockState(pos.down(height)).getBlock() == this){
+					height++;
+				}
+
+				if (height < 3) {
+					int age = state.get(AGE);
+					if (age == 15) {
+						world.setBlockState(pos2, this.getDefaultState());
+						BlockState state2 = state.with(AGE, 0);
+						world.setBlockState(pos, state2, 4);
+						state2.neighborUpdate(world, pos2, this, pos, false);
+					} else {
+						world.setBlockState(pos, state.with(AGE, age + 1), 4);
+					}
+
+				}
+			}
+			if (world.getBlockState(pos.down(1)).getBlock() == SpookyBlocks.TAINTED_SAND && random.nextInt(2) == 0) {
+				int height = 0;
+				int age = 0;
+				for(height = 0; world.getBlockState(pos.up(height)).getBlock() == this; height++) {
+					age = world.getBlockState(pos.up(height)).get(AGE);
+					world.setBlockState(pos.up(height), Blocks.AIR.getDefaultState());
+				}
+				SpookyCactusEntity entity = new SpookyCactusEntity(SpookyEntities.SPOOKY_CACTUS, world);
```

Comment:
I think we should have a newline before the entity just to split off the if-statement.